Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -94,6 +94,23 @@ extension Processor {
     assert(currentPosition >= start)
     assert(currentPosition <= end)
   }
+
+  mutating func reset(newBounds: Range<Position>)
+  {
+    self.bounds = newBounds
+
+    self.controller = Controller(pc: 0)
+    self.currentPosition = bounds.lowerBound
+
+    self.registers.reset(bounds.upperBound)
+    self.storedCaptures = self.storedCaptures.map {_ in .init()}
+
+    self.cycleCount = 0
```

Comment:
```suggestion
```

I'd say leave it  alone for now, the engine can query it. But we will probably want to keep metrics soon, including counting save points and restores, etc.